### PR TITLE
Add support for EventHub bindings

### PR DESCRIPTION
**Pull request by elprans**
Fixes: #119

### CODE REVIEW

File: tests/eventhub_functions/eventhub_trigger/function.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "scriptFile": "__init__.py",
+  "disabled": false,
+
+  "bindings": [
+    {
+      "type": "eventHubTrigger",
+      "name": "event",
+      "direction": "in",
+      "cardinality": "many",
```
Comment (pragnagopa) [inline tests/eventhub_functions/eventhub_trigger/function.json]: eventHubName is missing. Here is a sample function.json
```
      "type": "eventHubTrigger",
      "name": "myEventHubMessage",
      "direction": "in",
      "eventHubName": "samples-workitems",
      "connection": "AppSettingName",
      "consumerGroup": "$Default"
```

Comment (elprans) [reply tests/eventhub_functions/eventhub_trigger/function.json]: I tried that. With `eventHubName` I get:

```
fail: Host.Startup[0]
      Error indexing method 'Functions.eventhub_trigger'
Microsoft.Azure.WebJobs.Host.Indexers.FunctionIndexingException: Error indexing method 'Functions.eventhub_trigger' ---> System.InvalidOperationException: No event hub receiver named python-worker-ci
   at Microsoft.Azure.WebJobs.ServiceBus.EventHubConfiguration.GetEventProcessorHost(String eventHubName, String consumerGroup) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Extensions.EventHubs\EventHubConfiguration.cs:line 290
```

Comment (pragnagopa) [reply tests/eventhub_functions/eventhub_trigger/function.json]: You need to connect to an existing event hub. Here are the instructions : [event-hubs-create](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-create). When you create event hub trigger from portal, UI helps with choosing an existing eventhub or creating a new one.

Comment (elprans) [reply tests/eventhub_functions/eventhub_trigger/function.json]: I've done that. I have the connection string and everything. Doesn't help.

Comment (pragnagopa) [reply tests/eventhub_functions/eventhub_trigger/function.json]: I see. I will set up a C# event hug trigger example with the python-worker-ci eventhub and get back to you.

Comment (asavaritayal) [reply tests/eventhub_functions/eventhub_trigger/function.json]: I quickly checked with the C# and JavaScript examples. It seems that `path` and `eventHubName` have been used interchangeably to represent name of the event hub. See here: [event hub configuration](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-event-hubs#trigger---configuration)

Comment (asavaritayal) [reply tests/eventhub_functions/eventhub_trigger/function.json]: Adding @mhoeger to confirm while Pragna is out.

Comment (pragnagopa) [reply tests/eventhub_functions/eventhub_trigger/function.json]: @elprans  - Created EventHubTriggerCSharpTest in python-worker-ci-appveyor app. Here is the sample to trigger the function: https://github.com/soninaren/EventhubTrigger . Update eventHubName and connectionstring before running the sample

Comment (elprans) [reply tests/eventhub_functions/eventhub_trigger/function.json]: @pragnagopa Is there a working `function.json` example?

Comment (elprans) [reply tests/eventhub_functions/eventhub_trigger/function.json]: The `EventHubTriggerCSharpTest` still has `path` in its definition, which doesn't work with the latest host/extension:

```json
{
  "bindings": [
    {
      "type": "eventHubTrigger",
      "name": "myEventHubMessage",
      "direction": "in",
      "path": "pythonworkereventhubci",
      "connection": "pythonworkereventhub_RootManageSharedAccessKey_EVENTHUB",
      "consumerGroup": "$Default",
      "cardinality": "many"
    }
  ],
  "disabled": false
}
```

Comment (pragnagopa) [reply tests/eventhub_functions/eventhub_trigger/function.json]: Can you check EventHubTriggerCSharpTest ? Switch to Advanced editor for function.json

Comment (pragnagopa) [reply tests/eventhub_functions/eventhub_trigger/function.json]: Can you try
- change path to eventHubName
- Add app setting key: pythonworkereventhub_connection value: get connection string from the eventhub

**Comment by pragnagopa** [comment]
@soninaren - EventHubTrigger template on the portal still using "**path**"" for eventhub name. Also, connection string append entitypath. Is there a bug tracking this?

File: tests/eventhub_functions/eventhub_trigger/function.json (hunk 1 above)
Comment (elprans) [reply tests/eventhub_functions/eventhub_trigger/function.json]: How do I pass the app settings to the local host instance?

Comment (pragnagopa) [reply tests/eventhub_functions/eventhub_trigger/function.json]: set environment variable

**Comment by soninaren** [comment]
@pragnagopa and i validated that portal is using eventhubName for v2. In case eventHubName is present in function.json and entityPath is present in the connections string then connectionString gets precedence. Additionally eventHubName is still a required property in function.json even if it is overridden.

**Comment by pragnagopa** [comment]
@elprans -  python-worker-ci-appveyor app is a V1 app. you need to delete all the functions in it and opt into v2.

**Pull request merged**